docs/guides/circleci-setup.md

```
@@ -0,0 +1,668 @@
+# CircleCI Setup Guide
+
+This document provides comprehensive documentation for the CircleCI CI/CD pipeline configured for Hidetaka.dev.
+
+## Table of Contents
+
+- [Workflow Overview](#workflow-overview)
+- [Job Descriptions](#job-descriptions)
+- [Free Plan Optimizations](#free-plan-optimizations)
+- [Environment Variables](#environment-variables)
+- [Deployment Strategy](#deployment-strategy)
+- [Manual Cleanup](#manual-cleanup)
```

markdownlint-cli2 (0.18.1)

Link fragments should be valid

(MD051, link-fragments)